Update for Translate Japanese

saotak opened the pull request:
Added 4 files for Japanese version. Please find the PR!

saotak committed:
Japanese version for 5 pages.

saotak committed:
Japanese version for 5 pages.

JP_: Translation has done but some format is not fully optimized yet.

INPROGRESS_ : Translation is in progress.

saotak committed:
Removed Inprogress file

saotak committed:
Removed inprogress file

saotak committed:
Merge branch 'Translate_Japanese' of https://github.com/saotak/loopback.io into Translate_Japanese

saotak committed:
Added 4 new files

dhmlau commented on pages/ja/lb4/JP_Concepts.md:
```diff
@@ -0,0 +1,33 @@
+lang: ja
+title: 'キーコンセプト'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
```
sidebar should be `ja_lb4_sidebar` since it's Japanese.

dhmlau commented:
@saotak, thanks again for the translation. 

There are a few things we might need to take care first:
1. for all the japanese files, the sidebar should be `ja_lb4_sidebar`. 
2. permalink would be something like `/doc/ja/lb4/somefile.html` instead of `/doc/en/lb4/somefile.html`.
3. It seems like there are other files that are coming from the last PR. Could you please rebase the PR against `gh-pages` branch?
4. there's no need to rename the file name. 

Thanks!

saotak committed:
Modified some files

- sidebar into `ja_lb4_sidebar.`
- parmlik into `/doc/ja/lb4/somefile.html`
- file title

saotak commented:
Hi Diana, thank you for your reviewing the PR! I have modified the sidebar, permalink and file title.
As for your request 3., since I modified the sidebar and permalink in previous file, I just updated the previous 2 files too. So could you just check it again and accept this PR?

Thank you!

saotak committed:
Modified 2 previous files.

Re-added Concepts.md and README.
(The change in the file is only the permalink in Concepts.md.)

saotak committed:
filetitle

saotak committed:
Some translation modifications for the Concepts.md

saotak committed:
Update Crafting-LoopBack-4.md

dhmlau commented on pages/ja/lb4/Not-yet-translated.md:
```diff
@@ -1,10 +0,0 @@
----
-title: "Not yet translated"
```
@nabdelgadir, I think we need this page. Could you please confirm? Thanks.

nabdelgadir replied:
Yes, we still need this page for other pages that aren't translated

dhmlau commented on pages/ja/lb4/Concepts.md:
```diff
@@ -0,0 +1,33 @@
+---
+lang: ja
+title: 'キーコンセプト'
```
according to https://raw.githubusercontent.com/strongloop/loopback.io/gh-pages/pages/ja/lb4/Concepts.md, we need:
`layout: translation` in the front matter.  Similarly for the other pages.

saotak replied:
layout: translation

dhmlau commented on pages/ja/lb4/README_todo-tutorial.md:
```diff
@@ -0,0 +1,95 @@
+# @loopback/example-todo
```
This one is tricky because the English version is coming from the README file of [examples/todo](https://github.com/strongloop/loopback-next/blob/master/examples/todo/README.md). But _I think_ we can put the translated version in this location. 

@nabdelgadir, could you please confirm? In this case, we need to add the front matter for this page. And also rename it to `todo-tutorial.md` to match the original file name.

nabdelgadir replied:
I think there's already a todo tutorial in Japanese [here](https://loopback.io/doc/ja/lb4/todo-tutorial.html) where the README was copied.

dhmlau reviewed (commented):
@saotak, thanks for the changes. It's getting there. 
There are a few questions that we need some confirmation from @nabdelgadir since she's more familiar with the translated docs. 

I'd also like to point you to the instructions on previewing the content locally: https://github.com/strongloop/loopback.io#setup. In this case, you can also verify your changes on your side too. Thanks.

nabdelgadir commented on pages/ja/lb4/Crafting-LoopBack-4.md:
```diff
@@ -0,0 +1,358 @@
+---
+lang: jp
+title: 'LoopBack 4を構築する'
+keywords: LoopBack 4.0, LoopBack 4
```
Similarly here, we need `layout: translation`

saotak replied:
Added  `layout: translation`

nabdelgadir commented on pages/ja/lb4/Crafting-LoopBack-4.md:
```diff
@@ -0,0 +1,358 @@
+title: 'LoopBack 4を構築する'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: ja_lb4_sidebar
+permalink: /doc/jp/lb4/Crafting-LoopBack-4.html
```
`jp` -> `ja`

saotak replied:
Modified`jp -> ja`

nabdelgadir replied:
This one still shows `jp`
```suggestion
permalink: /doc/ja/lb4/Crafting-LoopBack-4.html
```

nabdelgadir commented on pages/ja/lb4/exposing-graphql-apis.md:
```diff
@@ -0,0 +1,116 @@
+title: 'GraphQL APIの公開'
+keywords: LoopBack 4.0, LoopBack 4, GraphQL
+sidebar: ja_lb4_sidebar
+permalink: /doc/jp/lb4/exposing-graphql-apis.html
```
Please add `layout: translation` and update the link from `jp` to `ja`

saotak replied:
Added and modified `layout: translation` and `jp to ja`.

nabdelgadir replied:
This is still `jp` in the permalink
```suggestion
permalink: /doc/ja/lb4/exposing-graphql-apis.html
```

nabdelgadir commented on pages/ja/lb4/express-with-lb4-rest-tutorial.md:
```diff
@@ -0,0 +1,327 @@
+title: 'Loopback REST APIでExpress Applicationを構築する'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: ja_lb4_sidebar
+permalink: /doc/jp/lb4/express-with-lb4-rest-tutorial.html
```
Same here

saotak replied:
Added and modified `layout: translation` and `jp to ja`.

nabdelgadir commented on pages/ja/lb4/index.md:
```diff
@@ -1,6 +1,5 @@
 ---
 lang: ja
-layout: translation
```
Please keep this line

saotak replied:
Recovered the line.

nabdelgadir reviewed (commented):
Thank you for your contribution! For each translated page, please update [the sidebar](https://github.com/saotak/loopback.io/blob/Translate_Japanese/_data/sidebars/ja_lb4_sidebar.yml) from `translated: false` to `translated: true`.

saotak committed:
Modification for #955

- sidebar
- layout: translation & jp >>ja

saotak commented:
@nabdelgadir , thank you for confirming the PR! I modified sidebar, translation tag and  jp>ja as I commented / committed. 
Regarding` README_todo-tutorial.md`, yes you're correct, it was the same contents as the todo tutorial. Shall I remove it?

Thank you!

saotak commented:
@dhmlau @nabdelgadir just for sure, let me confirm how to update the translation. Shall I just add the new translated files into [here](https://github.com/saotak/loopback.io/tree/Translate_Japanese/pages/ja/lb4
)?
Thank you again.

saotak committed:
Update for Todo tutorial and the sidebar

dhmlau commented:
@saotak, regarding translating the todo tutorial, if you can find the translated docs in https://loopback.io/doc/ja/lb4/todo-tutorial.html, there's no need to add. 

Please also see our docs for translating documentation: https://loopback.io/doc/en/contrib/Translating_articles.html. Thanks.

nabdelgadir commented:
> @dhmlau @nabdelgadir just for sure, let me confirm how to update the translation. Shall I just add the new translated files into [here](https://github.com/saotak/loopback.io/tree/Translate_Japanese/pages/ja/lb4)?
> Thank you again.

Yes there, and like @dhmlau said, https://loopback.io/doc/en/contrib/Translating_articles.html should help you with guidelines for translation. Thank you!

I also believe the Concepts page has already been translated https://loopback.io/doc/ja/lb4/Concepts.html. If it's missing any content can you add to it? But otherwise no need to include it in this pull request.

saotak commented:
@nabdelgadir Hi Nora, so sorry for my long absence since my project assignment has been changed. Thank you for your confirming the files. Yes as you suggested, Concepts.md has already been translated so I've just removed it.
Could you maybe marge the brunch now?

Thank you and hope you in safe!

saotak committed:
Removed Concept.md

nabdelgadir commented on pages/ja/lb4/Crafting-LoopBack-4.md:
```diff
@@ -0,0 +1,359 @@
+---
+lang: jp
```
```suggestion
lang: ja
```

nabdelgadir commented on pages/ja/lb4/README_todo-tutorial.md:
```diff
@@ -0,0 +1,95 @@
+# @loopback/example-todo
```
Don't forget to remove this file

nabdelgadir commented on pages/ja/lb4/Tutorials/todo/todo-tutorial-datasource.md:
```diff
@@ -0,0 +1,69 @@
+title: 'データソースを追加する'
+layout: translation
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
```
```suggestion
sidebar: ja_lb4_sidebar
```

nabdelgadir commented on pages/ja/lb4/Tutorials/todo/todo-tutorial-geocoding-service.md:
```diff
@@ -0,0 +1,248 @@
+title: 'ジオコーディングサービスと統合する'
+layout: translation
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
```
```suggestion
sidebar: ja_lb4_sidebar
```

nabdelgadir commented on pages/ja/lb4/Tutorials/todo/todo-tutorial-model.md:
```diff
@@ -0,0 +1,101 @@
+title: 'Todo Modelを追加する'
+layout: translation
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
```
```suggestion
sidebar: ja_lb4_sidebar
```

nabdelgadir commented on pages/ja/lb4/Tutorials/todo/todo-tutorial-putting-it-together.md:
```diff
@@ -0,0 +1,66 @@
+title: '統合する'
+layout: translation
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
```
```suggestion
sidebar: ja_lb4_sidebar
```

nabdelgadir commented on pages/ja/lb4/Tutorials/todo/todo-tutorial-repository.md:
```diff
@@ -0,0 +1,47 @@
+title: 'レポジトリを追加する'
+layout: translation
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
```
```suggestion
sidebar: ja_lb4_sidebar
```

nabdelgadir commented on pages/ja/lb4/Tutorials/todo/todo-tutorial-scaffolding.md:
```diff
@@ -0,0 +1,105 @@
+---
+lang: jp
```
```suggestion
lang: ja
```

nabdelgadir commented on pages/ja/lb4/Tutorials/todo/todo-tutorial-scaffolding.md:
```diff
@@ -0,0 +1,105 @@
+title: 'アプリの土台を構築する'
+layout: translation
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/jp/lb4/todo-tutorial-scaffolding.html
```
```suggestion
sidebar: ja_lb4_sidebar
permalink: /doc/ja/lb4/todo-tutorial-scaffolding.html
```

nabdelgadir commented on pages/ja/lb4/Tutorials/todo/todo-tutorial-controller.md:
```diff
@@ -0,0 +1,67 @@
+title: 'コントローラを追加する'
+layout: translation
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
```
```suggestion
sidebar: ja_lb4_sidebar
```

nabdelgadir commented on pages/ja/lb4/Not-yet-translated.md:
```diff
@@ -1,10 +0,0 @@
----
```
Remember to put this page back

nabdelgadir reviewed (commented):
Thank you @saotak for the update! There are still a few things left before we can merge. I put some as "Suggested changes" so it's easier to see where the change is needed. After these are fixed, please squash and rebase your commits. If you need help with squashing and rebasing, see [this documentation](https://loopback.io/doc/en/lb4/submitting_a_pr.html). Thanks!

saotak committed:
Update pages/ja/lb4/Crafting-LoopBack-4.md

Co-Authored-By: Nora <42985749+nabdelgadir@users.noreply.github.com>

saotak committed:
Update pages/ja/lb4/Crafting-LoopBack-4.md

Co-Authored-By: Nora <42985749+nabdelgadir@users.noreply.github.com>

saotak committed:
Update pages/ja/lb4/Tutorials/todo/todo-tutorial-datasource.md

Co-Authored-By: Nora <42985749+nabdelgadir@users.noreply.github.com>

saotak committed:
Update pages/ja/lb4/exposing-graphql-apis.md

Co-Authored-By: Nora <42985749+nabdelgadir@users.noreply.github.com>

saotak committed:
Update pages/ja/lb4/Tutorials/todo/todo-tutorial-geocoding-service.md

Co-Authored-By: Nora <42985749+nabdelgadir@users.noreply.github.com>

saotak committed:
Update pages/ja/lb4/Tutorials/todo/todo-tutorial-controller.md

Co-Authored-By: Nora <42985749+nabdelgadir@users.noreply.github.com>

saotak committed:
Update pages/ja/lb4/Tutorials/todo/todo-tutorial-scaffolding.md

Co-Authored-By: Nora <42985749+nabdelgadir@users.noreply.github.com>

saotak committed:
Update pages/ja/lb4/Tutorials/todo/todo-tutorial-scaffolding.md

Co-Authored-By: Nora <42985749+nabdelgadir@users.noreply.github.com>

saotak committed:
Update pages/ja/lb4/Tutorials/todo/todo-tutorial-model.md

Co-Authored-By: Nora <42985749+nabdelgadir@users.noreply.github.com>

saotak committed:
Update pages/ja/lb4/Tutorials/todo/todo-tutorial-putting-it-together.md

Co-Authored-By: Nora <42985749+nabdelgadir@users.noreply.github.com>

dhmlau commented:
@nabdelgadir, could you please review this PR and help landing it? Thanks!

nabdelgadir commented:
@saotak since there are only a few more comments, I'll update them and squash your commits for you.

nabdelgadir commented:
@saotak actually, I'm having some trouble with rebasing (it's including commits that are prior to this PR). Would you mind if I created a new PR with your changes and added you as a co-author?

saotak commented:
@nabdelgadir , my apologise for this really late response... Now we are in holiday in Japan so I can have some time to check the status.
> Would you mind if I created a new PR with your changes and added you as a co-author?
Of course, please do it whatever easy way for you!

nabdelgadir commented:
> @nabdelgadir , my apologise for this really late response... Now we are in holiday in Japan so I can have some time to check the status.

I've created https://github.com/strongloop/loopback.io/pull/980 as a replacement. 😄Would you mind sharing your email so I can add you as a co-author there? Thank you!

nabdelgadir commented:
Closing this PR as https://github.com/strongloop/loopback.io/pull/980 has landed. Thank you again! 😄

saotak commented:
@nabdelgadir @dhmlau, Thank you, noted for the pull request process as using Translate_Japanese branch.  And thank you so much for applying the translation!
